torch/csrc/jit/symbolic_script.cpp

```diff
@@ -46,23 +46,27 @@ const std::vector<std::string> functions = {
                 result = AD_unsqueeze_multiple(result, dim, n_dims)
             return grad * (self - result).exp()
 
-        def mean_0(self):
+        def mean_0(self, *, dtype: Optional[int]):
             self_size = self.size()
             self_numel = self.numel()
+            self_scalar_type = self.dtype
```

Comment:
this does not correctly unwrap the dtype I believe, `self_scalar_type` is still an `Optional[int]`, you will probably need to unwrap it using boolean refinement like `if is not None`

Reply:
I wasn't trying to unwrap, was just trying to be consistent with how the rest of the code is written. e.g. self.scalar_type() gets converted to self_scalar_type (`REPLACEMENTS` in `def saved_variables` in load_derivatives.py).

I think torch.mean() and to() both accept optional. If this is wrong though, how would I write a test that would catch it? It seems like this is working.

Reply:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L2704 hmmm that's pretty weird, from what i see in the schema, `to` does not take optional dtype, it also does not have default values, so I am not sure why this is working. 

For the mean, it looks like you are changing the API, can you add some tests to test the api in eager and see if Autodiff works in JIT? https://github.com/pytorch/pytorch/blob/master/test/common_methods_invocations.py#L339
The tests right now only have one test case, which does not contains the test case with `dtype`

Reply:
ah, I didn't notice this comment. Added a test of `('mean', (S, S, S), (), 'dtype', (True,), (), (), ident, {'dtype': torch.float64}),`.  Is that what you mean?

passed when I ran locally with:
python test_jit.py TestJitGeneratedAutograd.test_mean_dtype